### PR TITLE
MGMT-7401: Add cluster progress to ACI

### PR DESCRIPTION
**Pull request by slaviered**
# Assisted Pull Request

## Description
Expose the progress bar relevant information from the BE to KubeAPI

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -5560,7 +5560,22 @@ func (b *bareMetalInventory) GetClusterByKubeKey(key types.NamespacedName) (*com
...
+	//customize the host if it is bound to a cluster
+	var cluster *common.Cluster
+	if h.ClusterID != nil {
+		cluster, err = common.GetClusterFromDB(b.db, *h.ClusterID, common.SkipEagerLoading)
+		if err != nil {
+			return h, fmt.Errorf("can not find a cluster for host %s", h.ID.String())
+		}
+	}
+
+	err = b.customizeHost(&cluster.Cluster, &h.Host)
```
Comment (rollandf) [inline internal/bminventory/inventory.go:5577]: cluster can be nil here, no?

Comment (rollandf) [reply internal/bminventory/inventory.go:5577]: From CI:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2779fd7]

goroutine 1159 [running]:
github.com/openshift/assisted-service/internal/bminventory.(*bareMetalInventory).GetHostByKubeKey(0xc000734300, 0xc0024bf430, 0xe, 0xc0014e0330, 0x24, 0xc0014e0301, 0x24, 0x3856cc0)
	/go/src/github.com/openshift/origin/internal/bminventory/inventory.go:5555 +0x97
github.com/openshift/assisted-service/internal/controller/controllers.(*AgentReconciler).Reconcile(0xc0002b6540, 0x380b7d0, 0xc001590f30, 0xc0024bf430, 0xe, 0xc0014e0330, 0x24, 0xc001590f00, 0x0, 0x0, ...)
```

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -5560,7 +5560,24 @@ func (b *bareMetalInventory) GetClusterByKubeKey(key types.NamespacedName) (*com
 }
 
 func (b *bareMetalInventory) GetHostByKubeKey(key types.NamespacedName) (*common.Host, error) {
-	return b.hostApi.GetHostByKubeKey(key)
+	h, err := b.hostApi.GetHostByKubeKey(key)
+	if err != nil {
+		return nil, err
+	}
+
+	//customize the host as done in the REST API
```
Comment (rollandf) [inline internal/bminventory/inventory.go:5568]: Maybe add all the customize part inside the `if`? And handle all errors
```
	//customize the host as done in the REST API
	if h.ClusterID != nil {
		cluster, err = common.GetClusterFromDB(b.db, *h.ClusterID, common.SkipEagerLoading)
		if err != nil {
			return nil, fmt.Errorf("can not find a cluster for host %s", h.ID.String())
		}
  	       err = b.customizeHost(&cluster.Cluster, &h.Host)
		if err != nil {
			return nil, fmt.Errorf("failed to customize host %s", h.ID.String())
		}
	}
	return h, nil
```

Comment (slaviered) [reply internal/bminventory/inventory.go:5568]: No need. Event if the cluster is nil customize is still working on the host

**Pull request merged**